src/cogent3/parse/gff.py

```diff
@@ -45,46 +61,70 @@ def gff2_parser(f):
         if not line:
             continue
 
-        # parse columns
         cols = line.split("\t")
+        # the final column (attributes) may be empty
         if len(cols) == 8:
             cols.append("")
-        assert len(cols) == 9, line
-        (seqname, source, feature, start, end, score, strand, frame, attributes) = cols
+        assert len(cols) == 9, len(line)
+        (seqid, source, type, start, end, score, strand, phase, attributes) = cols
 
-        # adjust for python 0-based indexing etc.
+        # adjust for 0-based indexing
         (start, end) = (int(start) - 1, int(end))
         # start is always meant to be less than end in GFF
-        # and in v 2.0, features that extend beyond sequence have negative
-        # indices
+        # features that extend beyond sequence have negative indices
         if start < 0 or end < 0:
             start, end = abs(start), abs(end)
             if start > end:
                 start, end = end, start
-
-        # but we use reversal of indices when the feature is on the opposite
-        # strand
+        # reverse indices when the feature is on the opposite strand
         if strand == "-":
             (start, end) = (end, start)
 
-        # should parse attributes too
-        yield (
-            seqname,
-            source,
-            feature,
-            start,
-            end,
-            score,
-            strand,
-            frame,
-            attributes,
-            comments,
-        )
-
-
-def parse_attributes(attribute_string):
-    """Returns region of attribute string between first pair of double quotes"""
-    attribute_string = attribute_string[attribute_string.find('"') + 1 :]
-    if '"' in attribute_string:
-        attribute_string = attribute_string[: attribute_string.find('"')]
-    return attribute_string
+        if gff3:
+            attribute_parser = parse_attributes_gff3
+        else:
+            attribute_parser = parse_attributes_gff2
+        attributes = attribute_parser(attributes, (start, end))
+
+        rtn = {
+            "SeqID": seqid,
+            "Source": source,
+            "Type": type,
+            "Start": start,
+            "End": end,
+            "Score": score,
+            "Strand": strand,
+            "Phase": phase,
+            "Attributes": attributes,
+            "Comments": comments,
+        }
+        yield rtn
+
+
+def parse_attributes_gff2(attributes, span):
+    """Returns a dict with name and info keys"""
+    name = attributes[attributes.find('"') + 1 :]
+    if '"' in name:
+        name = name[: name.find('"')]
+    id_ = name if name else str(span)
```

Comment:
span not safe as multiple distinct features can map to same coords. We need to construct a fake identifier with a count.

```python
class gff_attr_parser:
      def __init__(self, gff3):
        # define attributes
      self_num_fake = 0
       self._gff3 = gff3
    def __call__(self, attrs):
       if self._gff3:
        pass
```